charms/openfga-k8s/src/charm.py

```diff
@@ -184,20 +185,36 @@ def _on_update_status(self, _):
         self._ready()
 
     @requires_state
-    def _create_token_secret(self, event):
-        if not self._state.token_secret_id:
+    def _create_token(self, event):
+        if JujuVersion.from_environ().has_secrets:
```

Comment:
Can't we rewrite it this method to avoid repetitions?

Reply:
done, i think.. at least improved a bit